Ginger/GingerCoreCommon/VariablesLib/VariableSelectionList.cs

```diff
@@ -121,14 +124,53 @@ public override void ResetValue()
         {
             if (OptionalValuesList.Count > 0)
             {
+                OptionalValuesList.CurrentItem = OptionalValuesList.ElementAt<OptionalValue>(0);
                 Value = OptionalValuesList[0].Value;
             }
 
         }
 
-        public override void GenerateAutoValue()
+        public override void GenerateAutoValue(ref string errorMsg)
         {
-            //NA
+            if (OptionalValuesList.Count == 0)
+            {
+                Value = string.Empty;
+                errorMsg = "Generate Auto Value is not possible because Selection List is empty";
+                return;
+            }
+
+            int index = OptionalValuesList.IndexOf((OptionalValue)OptionalValuesList.CurrentItem);
```

Comment:
need to look for index by the value and not by CurrentItsm which may not updated. use optionalValuesList.IndexOf(Value)